inference/worker/PLUGINS.md

```diff
@@ -0,0 +1,68 @@
+# Plugin system for OA
+
+This is a basic implementation of support for external augmentation and
+OpenAI/ChatGPT plugins into the Open-Assistant. In the current state, this is
+more of a proof-of-concept and should be considered to be used behind some
+experimental flag.
+
+## Architecture
+
+There is now some kind of middleware between work.py(worker) and the final
+prompt that is passed to the inference server for generation and streaming. That
+middleware is responsible for checking if there is an enabled plugin in the
+userland/UI and if so, it will take over the job of creating curated pre-prompts
+for plugin usage, as well as generating subsequent calls to LLM(inner
+monologues) in order to generate the final externally **augmented** prompt, that
+will be passed back to the worker and next to the inference, for final LLM
+generation/streaming tokens to the frontend.
```

Comment:
Right now I am just trying to fully comprehend your implementation concept here so forgive me if I am wrong but the **augmented** prompt being passed to the worker for final LLM generation of tokens means all heavy lifting is done at the prompt preprocessing stage. Therefore does this imply the final role of the generated tokens is simply to "phrase" the output in an organic language/text for humans or summarize in some way?

Let me elaborate. I really like the approach of logical reasoning/sequence here that helps the model make more accurate conclusions. But once it has the final prompt and is supposed to come up with a final answer ("after doing the research"), what triggers it to generalize over all previous observations? For instance, I am imagining a multi-step calculation or a sequence of queries providing different possibly overlapping pieces of information. So perhaps I am missing this in the draft changes around the final prompt but it seems to me the LLM is asked to just continue a long but limited chat chain of collected and delimited observations as well as some "use_tool: no" lines?